### ISSUE TITLE
How to differentiate the use of a classification term between two objects once loaded into a graph?

### ISSUE THREAD

**Issue by beaudet** [linked-art/linked.art#461]
I feel like I'm overlooking something pretty basic with this question, but if I have the following term applied to an art object as follows:

      "id": "http://vocab.getty.edu/aat/300033618",
      "type": "Type",
      "_label": "Painting",
      "classified_as": [
        {
          "id": "http://vocab.getty.edu/aat/300435443",
          "type": "Type",
          "_label": "Type of Work"
        }
      ]

and one of the parts for that object (or for that matter, another object in the graph), defines another classification for the same term, e.g.

      "id": "http://vocab.getty.edu/aat/300033618",
      "type": "Type",
      "_label": "Painting",
      "classified_as": [
        {
          "id": "http://vocab.getty.edu/aat/300053001",
          "type": "Type",
          "_label": "Process and Techniques"
        }
      ]

How does one distinguish these differences in a knowledge graph which would presumably accumulate all of the classifications for the term under the node for the term rather than with the object where the sub-classifications are defined?

in other words, after loading the JSON-LD, there's now a node in the graph with URI

http://vocab.getty.edu/aat/300033618

which has two sub-classifications (Type of Work + Process & Techniques) hanging off of it.

With SHACL validation, so long as ONE of the two uses of the term defines "Type of Work", no warning is issued since the required sub-classification for the object part is already present in the graph since it was defined by the parent.

In general, if the IDs of terms are used directly in the graph rather than blank nodes, how does one go about differentiating between use of a term in one object rather than another?

**Comment by beaudet** [comment]
An example graph in JSON-LD with the corresponding Turtle serialization showing how the data is accumulated in the graph, i.e.

**<http://vocab.getty.edu/aat/300033618>
        rdf:type         crm:E55_Type ;
        rdfs:label       "Painting" ;
        crm:P2_has_type  <http://vocab.getty.edu/aat/300138075> , <http://vocab.getty.edu/aat/300435443> .**

SOURCE JSON-LD
```
{
  "@context": "https://linked.art/ns/v1/linked-art.json",
  "id": "https://linked.art/example/object/parent",
  "type": "HumanMadeObject",
  "_label": "Example Painting",
  "classified_as": [
    {
      "id": "http://vocab.getty.edu/aat/300033618",
      "type": "Type",
      "_label": "Painting",
      "classified_as": [
        {
            "id": "http://vocab.getty.edu/aat/300435443",
            "type": "Type",
            "_label": "Type of Work"
         }
      ]
    }
  ],
  "part": [
    {
      "type": "HumanMadeObject",
      "id": "https://linked.art/example/object/child",
      "_label": "Example Painting using Painting term used in the context of a technique",
      "classified_as": [
        {
          "id": "http://vocab.getty.edu/aat/300033618",
          "type": "Type",
          "_label": "Painting",
          "classified_as": [
            {
              "id": "http://vocab.getty.edu/aat/300138075",
              "type": "Type",
              "_label": "Processes and Techniques"
            }
          ]
        }
      ]
    }
  ]
}
```

AS TURTLE
```
@prefix crm:  <http://www.cidoc-crm.org/cidoc-crm/> .
@prefix owl:  <http://www.w3.org/2002/07/owl#> .
@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd:  <http://www.w3.org/2001/XMLSchema#> .

<http://vocab.getty.edu/aat/300033618>
        rdf:type         crm:E55_Type ;
        rdfs:label       "Painting" ;
        crm:P2_has_type  <http://vocab.getty.edu/aat/300138075> , <http://vocab.getty.edu/aat/300435443> .

<https://linked.art/example/object/child>
        rdf:type         crm:E22_Human-Made_Object ;
        rdfs:label       "Example Painting using Painting term used in the context of a technique" ;
        crm:P2_has_type  <http://vocab.getty.edu/aat/300033618> .

<https://linked.art/example/object/parent>
        rdf:type                crm:E22_Human-Made_Object ;
        rdfs:label              "Example Painting" ;
        crm:P2_has_type         <http://vocab.getty.edu/aat/300033618> ;
        crm:P46_is_composed_of  <https://linked.art/example/object/child> .

<http://vocab.getty.edu/aat/300435443>
        rdf:type    crm:E55_Type ;
        rdfs:label  "Type of Work" .

<http://vocab.getty.edu/aat/300138075>
        rdf:type    crm:E55_Type ;
        rdfs:label  "Processes and Techniques" .
```

AS RDF
```
<rdf:RDF
    xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
    xmlns:owl="http://www.w3.org/2002/07/owl#"
    xmlns:crm="http://www.cidoc-crm.org/cidoc-crm/"
    xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema#">
  <crm:E22_Human-Made_Object rdf:about="https://linked.art/example/object/parent">
    <rdfs:label>Example Painting</rdfs:label>
    <crm:P46_is_composed_of>
      <crm:E22_Human-Made_Object rdf:about="https://linked.art/example/object/child">
        <rdfs:label>Example Painting using Painting term used in the context of a technique</rdfs:label>
        <crm:P2_has_type>
          <crm:E55_Type rdf:about="http://vocab.getty.edu/aat/300033618">
            <rdfs:label>Painting</rdfs:label>
            <crm:P2_has_type>
              <crm:E55_Type rdf:about="http://vocab.getty.edu/aat/300138075">
                <rdfs:label>Processes and Techniques</rdfs:label>
              </crm:E55_Type>
            </crm:P2_has_type>
            <crm:P2_has_type>
              <crm:E55_Type rdf:about="http://vocab.getty.edu/aat/300435443">
                <rdfs:label>Type of Work</rdfs:label>
              </crm:E55_Type>
            </crm:P2_has_type>
          </crm:E55_Type>
        </crm:P2_has_type>
      </crm:E22_Human-Made_Object>
    </crm:P46_is_composed_of>
    <crm:P2_has_type rdf:resource="http://vocab.getty.edu/aat/300033618"/>
  </crm:E22_Human-Made_Object>
</rdf:RDF>
```
```

**Comment by beaudet** [comment]
Once we have member_of collections, the meta-type assertion should be used to determine which collection to validate in which collection the term should exist.  e.g. "type of work" term means look in the collection of "types of work" to see if the parent term exists there.

**Comment by azaroth42** [comment]
Yeah, ... don't do that :) Either painting should be the object type, or the technique, but not both.

**Comment by beaudet** [comment]
acknowledged, but regardless of the motivation it seems likely that a classified_as term will be metatyped multiple times and that breaks the ability to round-trip the data in a way that will reconstitute only the metatypes applicable to object where those metatypes are used.  In other words, the context where the metatype is being applied isn't really on the term itself, but on the application of that term to the linked art entity.  I think we discussed this on the call, but it almost seems like each use of classified_as involving metatypes should mint a new identifier to capture the application of the term to the linked art entity so that metatypes will be scoped properly.  Right now, it's a gray area.  There can be contradictory metatypes that are valid data representations, but make no logical sense and there will be aggregation of metatypes collected for each term that when exploded in the JSON would, I imagine, duplicate those aggregated terms everywhere the term is used.

**Comment by azaroth42** [comment]
The same problem exists anywhere we use vocabulary or any other sort of shared identities with a more specific ontology. You say that the museum is a Place, I say that it's a Group, and someone else is thinking of the physical building... if you aggregate all three classes associated with the same URI into a single graph, you have a huge mess ... but if the URI is ambiguous as to which of those it describes, none of us are individually wrong.

As far as I can tell, the possible solutions are worse than the symptoms:

1. To manage our own vocabulary and/or make decisions about how existing terms must be used in the context of linked art. For the original example, we could declare that aat:300435443 must only be used with classified_as on a HumanMadeObject, and that aat:300053001 must only be used as the `technique` (per #561) of an activity. In aggregate that would be an enormous task as we'd be rebuilding AAT.

2. Come up with new relationships for every potential reuse of a term, e.g. `nationality`, `gender`, `type_of_work`, `type_of_part`, `style`, `genre`, `shape, etc etc all would need new relationships. This is why we went to metatypes in the first place.

3. Use attribute assignments everywhere to avoid direct assertions. As with any reification approach, it adds a lot of overhead and complexity. 

4. Named Graphs. But then we're replacing one technology problem (triplestores) with another (quadstores). 

Creating a new entity every time seems even worse, as there's no way to know where in the big wide world people have used and misused these URIs, and when that data might come into the triplestore. Especially as it's not just metatypes, but basically any assertion. Consider the question about `_label` (#539) and the `type` of a resource in my opening :(

We can of course discuss possible solutions, but I don't know of any that don't have huge overheads.

**Comment by beaudet** [comment]
I think we should talk about possible solutions to maintain the integrity of the graph to be able to round-trip it to RDF.  That almost seems like a basic expectation to me given that one of the first things data consumers will probably do is load it into a graph database and run SPARQL against it.  I'm not speaking from a place of much experience at all on this, but it doesn't quite feel right to me that meta-statements made in the context of a term's relationship to a single entity would (not so obviously) bleed into any entity using that term since the metadata is for the term the way it's currently expressed rather than a blank node that also has a relationship with the term.  I could understand if the use of metatypes is clearly and intentionally about making global graph statements about a term, but I would argue in that case that the terms should be their own entities and the metatypes would be applied at the entity level not embedded in the object data.  But I don't think that's the case. 

In other words, any linked art entity can be classified with terms, but rather than a direct classification, there's a node that encapsulates two relationships.  So, to me at least, it seems like (3) is the most preferable / viable solution.  

HumanMadeObject -> attributeAssignment -> [ classified_as -> term URI, usage_context -> term URI ]

Maybe a pattern like this could be mandatory when specifying the term's usage context, but if there's no context to convey, it falls back on the standard pattern (if there's an elegant way to do that).

**Comment by azaroth42** [comment]
To try and be clear about my opinion:

The problem is not meta-typing, the problem is at the level of the data. If the data was clean, and there were sufficient and clear vocabulary terms for all of the situations, then we wouldn't have the problem, because the metatypes would never collide. The metatypes are there to ensure that consuming systems don't need to understand the impossibly long list of all types of object, types of work, nationalities, occupations, and so on, but instead understand the metatype to know what sort of classification it is. So, in my opinion, following the same principle as solving API problems in the API, model problems in the model, we should solve data problems in the data.

I think that metatypes do follow this principle: we're solving a problem with the model/ontology (lack of clarity around type of classification) in the model/ontology (with a metatype, which just reuses the existing classification pattern, and without introducing new terms).

It could also be argued that it's a technology problem -- that the issue is with the use of a triplestore as the system of record. With LUX, we don't run into the problem because we don't use a triplestore as the system of record, and we don't use SPARQL as our primary interaction with the data. Any client that uses the JSON as JSON (which I think will be the vast majority of them) will not care either. So implementations using triplestores need to be aware of the problem and to work around it ... but I don't believe that should be a cost for everyone to have to pay.

On the technology side, I also consider Linked Art to be an interoperability format or exchange / interchange format, not a database schema. To bastardize Postel's law: be liberal with what you accept, and strict with what you ingest into your system. Linked Art isn't the schema for TMS or EMu or Museum+ or any other core system of record. To ingest a Linked Art record into TMS would be a much bigger lift than a Linked Art record into a triplestore to avoid this particular issue. The same solution, some data transformation to suit the technological requirements, would be relatively simple ... just not part of the specification.

For the object type / technique case for painting, it can be solved by cleaning up the data, as there are two separate AAT terms for painting (object type) - aat:300033618 -  and painting (process) - aat:300054216. Granted that is not always the case, and sometimes there's one or the other but not both. That is a problem with the data and vocabulary layer, not with the model layer.

e.g.
<img width="501" alt="461-diagram" src="https://github.com/linked-art/linked.art/assets/871868/562568d1-9224-4fe2-8496-787128e0c5e8">

So, if we accept the above, the solution at the data layer or technology layer is "It hurts when I do this ... Don't Do That Then!" but how can we make it easier to not do the painful thing?

* Data Documentation. When we run into the problem in real data, then document the solution. Work with Getty Vocabs to ensure that there are appropriate terms in AAT to cover the various requirements, and if not, then mint our own.
* Vocabulary analysis and Documentation. We can spend more time with #186 to be much clearer around the expected use of terms.
* Technology Documentation. Highlight the challenge in the documentation so implementers using triplestores are aware of the problem. We can come up with a non-interchange solution (likely per Dave around reification) that can be written up, coded up, and shared without adding complexity to the baseline case of JSON-as-JSON.
* Validators. Create (per SHACL validation) graph based validators as well as syntax level JSON validators to detect these things.

**Comment by azaroth42** [comment]
Copying the relevant `type` example from #560:

Some URIs in LUX have the same equivalent with different classes, for example:

* https://lux.collections.yale.edu/view/group/de48ab54-2bb4-4334-be20-4c2912aae220 is the University of Cambridge as a Group.
* https://lux.collections.yale.edu/view/place/014bc020-5b74-4cde-8a72-e7465a3bc09d is the University of Cambridge as a Place.
* And both have https://www.wikidata.org/wiki/Q35794 as an equivalent, meaning that we assert:

```
wd:Q35794 rdf:type crm:E74_Group .
wd:Q35794 rdf:type crm:E53_Place .
```

Which is of course impossible in the CRM and would make a terrible mess of a triplestore.

We could discuss a URI level solution, but one which would make interoperability Very Hard at the graph level: Add structured fragments to the end of the URIs to keep them separate in the graph.

e.g. something like:

```
wd:Q35794#la:quaGroup rdf:type crm:E74_Group ;
  la:specializationOf wd:Q35794 .
wd:Q35794#la:quaPlace rdf:type crm:E53_Place ;
  la:specializationOf wd:Q35794 .
```

or for the original use case:

```
aat:300033618#la:quaTechnique rdf:type crm:E55_Type .
```

**Comment by beaudet** [comment]
What about using classified_as whenever there's no metatype in play and a different class carrying the metatyping and a term when there is? That would ensure the scope of metatype assertions are limited to the entity being described while enabling a more free-form handling of classified_as data.  I'm imagining how an application would make use of both cases.  For classified_as it might collect all terms in the result set and try to maintain the relationships between those terms if they are in the same section of a naming authority's hierarchy to infer the metatype.  For example, terms appearing under a parent term labeled "types of objects" would be relatively straightforward to unwind and dispensing with metatypes is probably fine.  For other situations where application of a term is more semantically nuanced, an Attribute Assignment or equivalent would carry the metatype and all the metatypes could be gathered up together to stitch the data together.  That sounds feasible and not particularly complicated and it also would enable round-tripping of data.

**Comment by azaroth42** [comment]
Not sure that I follow, sorry. Can you write out an example? Changing the relationships won't help -- if there's any relationship with the same URI as the subject, then it's going to collide due to the nature of triplestores not having any graph boundaries.

The triplestore-friendly structure would be something that asserted:

In the context of <object>, the term <painting> is <classified_as> a <type of work>.

or:

```
{
  "type": "HMO",
  "local_assertions": [
    {"assigned": "aat:type-of-work", "assigned_to": "aat:painting", "assigned_property": "classified_as"}
  ]
}
```

And then the query would need to check the document URI, and the assigned_to Type's URI and the assigned_property, to fetch the value of assigned.

**Comment by beaudet** [comment]
Sure!  (and in full disclosure I iterated with GPT on this response).

**Current Linked Art Metatyping Pattern Example**
_Female as Assigned Gender:_

{
  "@context": "https://linked.art/ns/v1/linked-art.json",
  "id": "https://linked.art/example/person/12",
  "type": "Person",
  "_label": "Person A",
  "classified_as": [
    {
      "id": "http://vocab.getty.edu/aat/300189557", // Female
      "type": "Type",
      "_label": "Female",
      "classified_as": [
        {
          "id": "https://homosaurus.org/v3/homoit0000078",
          "type": "Type",
          "_label": "Assigned Gender"
        }
      ]
    }
  ]
}
Female as Gender Identity:

json
Copy code
{
  "@context": "https://linked.art/ns/v1/linked-art.json",
  "id": "https://linked.art/example/person/13",
  "type": "Person",
  "_label": "Person B",
  "classified_as": [
    {
      "id": "http://vocab.getty.edu/aat/300189557", // Female
      "type": "Type",
      "_label": "Female",
      "classified_as": [
        {
          "id": "https://homosaurus.org/v3/homoit0000571",
          "type": "Type",
          "_label": "Gender Identity"
        }
      ]
    }
  ]
}
Issue in Graph Databases

When these data entries are loaded into a graph database, the context for the term "Female" becomes blurred. In a graph database, the term "Female" is associated with both "Assigned Gender" and "Gender Identity" metatypes, but the database doesn't maintain the distinct association with each individual (Person A and Person B). This leads to a situation where the gender identity and assigned gender of both artists become unclear or falsely stated. The term "Female" gets aggregated across the dataset without maintaining the individual-specific context, leading to potential misinterpretation and loss of data integrity.

This ambiguity underscores the need for a data modeling approach that maintains the specificity and context of each term's application, particularly in a graph database environment. The proposed attribute assignment method addresses this by explicitly tying each term to its intended context within the scope of each individual entity, thereby preserving the clarity and accuracy of the data.

**Comment by azaroth42** [comment]
Ahh, thanks Dave! (and ChatGPT :D)

Agreed that moving the metatype classification on to the attribute assignment solves the problem for metatypes, as the AA is a blank node that's never reused, so there's no collisions with the underlying AAT term.

The current metatypes:

* Type of Object (on HMO, DigObj)
  * Type of Part (on HMO, DO that are parts of some other HMO/DO) ... but this could be dropped in favor of always using Type of Object, as part-ness is contextual in itself
* Type of Work (on VI, LO)
* Style (VI/LO)
* Shape (HMO)
* Type of Statement (on all our `referred_to_by` statements everywhere)
* Nationality (Person, Group)
* Occupation / Role (Person, Group)
* Gender (Person)

The same technique (reification) would also solve the other collisions (e.g. the class for 'pencil' in `object isA Type(pencil)` vs `object made_of Material(pencil)`) but at great usability expense, especially for the extremely common case of human readable statements.

```
{
  "referred_to_by": [
    {
      "type": "LinguisticObject",
      "content": "Oil on Canvas",
      "attributed_by": [
        {
          "type": "AttributeAssignment",
          "assigned_property": "classified_as",
          "assigned": [
            {
              "id": "http://vocab.getty.edu/aat/materials",
              "type": "Type",
              "_label": "Material Statement"
            }
          ],
          "classified_as": [
            {
              "id": "http://vocab.getty.edu/aat/300418049",
              "type": "Type",
              "_label": "Type of Statement"
            }
          ]
        }
      ]
    }
  ]
}
```

Given the short list of metatypes and their ubiquity in data, I think I'd be happier with minting our own specializations of `P2_has_type`

```
{
  "referred_to_by": [
    {
      "type": "LinguisticObject",
      "content": "Oil on Canvas",
      "type_of_statement": {
        "id": "http://vocab.getty.edu/aat/materials",
        "type": "Type",
        "_label": "Material Statement"
      }
    }
  ]
}
```

and then documenting how to get from `la:type_of_statement` to the attribute assignment pattern for graph based processing using only CRM properties.

**Comment by beaudet** [comment]
Sure, that works, although I thought one disadvantage of the mint new relationships approach is:
`Come up with new relationships for every potential reuse of a term, e.g. nationality, gender, type_of_work, type_of_part, style, genre, `shape, etc etc all would need new relationships. This is why we went to metatypes in the first place.`
but if the number is relatively small, I think that's the clearest approach.  

Should we also recommend avoiding metatyping?  From a practical perspective that seems more likely to create annoyances / inconveniences down the road.  For example, imagine creating an AI agent that is trained on the use SPARQL and the linked art model and knows how to formulate specific sparql queries in response to a natural language search.  It seems much more feasible to do that with SPARQL in a no-code way as opposed to juggling JSON.

**Comment by azaroth42** [comment]
My preference is still the status quo (metatypes) and document how to avoid the issue when managing a graph, as it's not just metatypes that have the problem, we also see real world cases of `type` and `_label` collisions. 

The underlying cause is making assertions (`type`, `_label`, and `classified_as`) about other institutions' URIs and then aggregating those assertions from different sources. If you're inconsistent internally, well, that's your problem :) But even if you're consistent internally about others' data, when your data is merged with someone else who is differently internally consistent, the problem returns.

So, to return to the possible solutions as I see them.

* Reification. Attribute Assignments everywhere, per your comment. (Doesn't solve the label or type issue, and very verbose)
* New Properties. Devolve the AA into a property. (ditto, but trades verbosity for a non-interoperable approach outside Linked Art)
* Named Graphs. Replaces one technology problem with another, and probably doesn't solve everything anyway.
* New URIs. We could manage our own URI space when collisions are detected, or beg Patricia for new URIs in AAT. Doesn't solve non AAT problems.
* Modified URIs. Per my comment a few up, we could have a convention of adding fragments to external URIs to align them with the usage to avoid collisions. This solves the problem, but would be very costly to apply across the board, and breaks any hope of inter-graph connectedness. A slight variation on New URIs to have them minted by convention everywhere, rather than by authority in one place.
* Status Quo. Document the challenge, document the possible solutions, and have adopters pick their internal flavor separately from external interoperability flavor.

Other than the status quo, my second choice would be new properties as they're easy to use and understand in most implementations, and use cases that require interoperability with non Linked Art profile CRM data are going to have to do some transformation work regardless.

**Comment by azaroth42** [comment]
To try to catalog the places we use external URIs as the subject of triples:

* References to anything
  * equivalent
  * about
  * represents
  * influenced_by

* References to AAT / Concepts
  * classified_as
  * technique
  * broader
  * made_of (Material)
  * unit (MeasurementUnit)
  * language (Language)
  * currency (Currency)

* References to TGN / Places
  * part_of
  * took_place_at
  * residence
  * current_location

* References to ULAN / People or Groups
  * carried_out_by
  * participated_in
  * member_of (Group)
  * current_owner
  * transferred_custody|title_to|from

* Misc References
  * access_point can point to anything (and asserts that it's a digital object)
  * conforms_to 

For the references to specific types, we could solve the `type` and `_label` problems by only using the URI. e.g. `"carried_out_by": ["ulan:rembrandt", "https://local.data/person/12345"] But then we lose most of the benefit of `_label` as a way to associate some name with the otherwise opaque URI. 

For the non-specific typed references, equivalent is always the same class as the thing which has the equivalent property, but `about`, `represents` and `influenced_by` can point to anything. It seems like having access to the class should help implementations know how to process the link, and might even be necessary for it. Similarly for Person/Group it could be valuable to know the class.  Given the API should be consistent (e.g. if it's an array of strings, then it's an array of only strings, not sometimes also json objects), I think this would be throwing out far too much value.

**Comment by azaroth42** [comment]
Okay ... how about this ...

We decided to allow `equivalent` in a reference in #439, and I wonder if we can double down on that to solve this issue?

The problem is that we don't want to make conflicting assertions about other institutions' URIs. e.g. that aat:pencil is sometimes a material, and sometimes an object type. Or that female is sometimes a gender expression, and sometimes a biological trait. 

We could go further than #439 that allowed adding `equivalent` to say that the reference might _only_ have `equivalent` and not also `id`. Let's say it's the entirely vanilla case of metatypes on statements. Instead of saying that aat:material-statement is classified_as a aat:brief-text, we could say that there's a blank node (or a locally identified node with a URI) that has an equivalent of aat:material-statement which is classified_as aat:brief-text

So, instead of
```
"referred_to_by": [
  {
    "type": "LinguisticObject",
    "content": "Oil on Canvas",
    "classified_as": [
      {
        "id": "aat:material-statement",
        "type": "Type",
        "classified_as": [
          {
            "id": "aat:brief-text",
            "type": "Type"
          }
        ]
      }
    ]
  }
]
```

It would be:
```
"referred_to_by": [
  {
    "type": "LinguisticObject",
    "content": "Oil on Canvas",
    "classified_as": [
      {
        "type": "Type",
        "equivalent": [
          {
            "id": "aat:material-statement",
            "type" : "Type"
          }
        ],
        "classified_as": [
          {
            "id": "aat:brief-text",
            "type": "Type"
          }
        ]
      }
    ]
  }
]
```
Which doesn't conflict with anything anyone else says, and doesn't add annoying reification. The cost is one more level of join in a query (you'd have to search for all the bnodes with an equivalent, rather than the id directly), but that can be mitigated by assigning a local URI for material-statement and using that instead.

**Comment by beaudet** [comment]
Bravo, there's no longer a round tripping problem nor an "authorship" conflict since external IDs are not modified. 

Furthermore, rather than saying that material statements are brief texts, the facts of which could easily be disputed in a global context, you're saying what exactly?  I missed the Equivalent discussion, but from the data, I would guess you're saying that something is referred to by the text content "Oil on Canvas" which is a brief text and equivalent to a material statement. 

Is there a reason not to just classify the linguistic object also as a material statement?  There's a lot of pragmatic value that might result from keeping the classifications all at the same level and treating them as "aspects" that enable certain functionality in an application to apply to an object.  Using equivalent doesn't introduce too much complexity in that respect, but I'm missing the importance of using equivalent rather than classified_as.

**Comment by beaudet** [comment]
Regarding the question posed for the meeting agenda tomorrow on whether mandatory linked art URIs for vocabulary should be minted, I would say yes because I think it would be pretty helpful and there probably aren't THAT many of them.  We might even be able to use GitHub to serve those up or at least drive the workflow via pull requests that results in automated publishing to the linked art site.  That way, we can add them in near real-time to the vocabulary without having to increment the specs.  Any graphs using blank nodes as recommended could just pull up a list of the equivalent usages to figure out where they might have new concepts that need to be discussed and published as LA vocabulary URIs.

**Comment by cbutcosk** [comment]
I see the need for _some_ solution here. @beaudet's gender example reads as a pretty definitive demolishment of the status quo re: graph-round tripping of meta-types, and note this is a problem in pure JSON serializers too if they use a kv-based cache layer (a common feature of out-of-band taxonomy enrichers). 

That said in most of the cases where users would want to multiply-metatype `aat:female` I would advise them as @azaroth42 did upthread--that's a new term with its own identity, so it is probably clearer and more manageable to use a new identifier. In the case of domains with inequitable term distribution like gender this seems especially desirable: _don't you want other people to use that great new term_? 

Using a blank node is a choice to hide a node's identity semantics from the user. That's fine in cases like Dave's where he wants to partition those terms without making a URI for whatever reason (easy to imagine that at the low-end of implementations too, where infra complexity might be too much to mint a URI). But (AFAIU) JSON-LD [specifies how to identify blank nodes but does not specify an algorithm for blank node generation and deduplication](https://w3c.github.io/json-ld-syntax/#identifying-blank-nodes) so--buyer beware.

Anyway long way of saying since I can't make the meeting today: +1 to minting term identifiers, this seems more equitable than leaving users and implementers at the mercy of the GVP and Wikidata anyway. +1 to scrutinizing this new pattern a bit, though if more capacious terms are emitted that seems less crucial.

**Comment by beaudet** [comment]
Great discussion on the call today.  I keep thinking this is primarily a data problem.  What are the disadvantages of an approach where L.A. version 1.0 allows for both the status-quo metatyping pattern and the equivalence proposal in the API but that we label the pre-1.0 metatyping pattern as deprecated?  That would put mean a translation layer would be required only by systems interacting with the pre-1.0 pattern?

Are there challenges using the equivalence pattern that we should consider, either using the JSON-LD directly or in a triple store or is it relatively straightforward to make use of that?

**Comment by azaroth42** [comment]
To try and summarize the pros and cons from the [call](https://docs.google.com/document/d/1SctzEqd4PaaRisLA4Y8bXAyLNW2O6EPFCxv9gz40L8Q/edit):

* It is a real challenge that has been encountered in the wild by multiple organizations, in different situations
* Adopters will expect to be able to load LOD into a triplestore and not generate garbage by following the specification -- a reputation and usability hit, as they also likely won't read the documentation
* Documenting the differences for SPARQL queries between the model and the local implementation is additional work that can be minimized if the model doesn't introduce necessary local modifications

* The challenge is at the intersection of technology and data -- the model works just fine if there are unique URIs for different concepts (etc) -- and should thus be solved at one of those two levels
* The use of `equivalent` is good, and the notion of it being a proxy is a pattern that's used in other specifications
* Mandating `equivalent` rather than `id` is seen as a step too far, as using AAT (etc) is what people expect to be able to do
* The transformation (move `id` to `equivalent` on load) can be documented and is a small number of lines of code, that doesn't need access to more than a single JSON document at a time. Most systems will need some degree of code to load data, even if that's only expanding the JSON-LD into triples. Easy to document, and the code can be provided, and even available as a web service.

Outstanding question is: Should no `id` and `equivalent` be legal in the API? (e.g. this blank node is a proxy for the entities given in `equivalent`)

**Comment by azaroth42** [comment]
IMO, it shouldn't be legal for the following reasons:

* Freedom from choice -- there should be one way to do things, as much as possible. 
* Cost is high if we allow it, as every processor will need to check both `id` and `equivalent/id` for everything. Any path (in json or sparql) that goes to or through a reference would need to process both. That's a lot of joins.
* Solve at the right layer -- this is promoting a fix through the model and up to the API from the technology layer. 
* Makes determining a reference vs an embedded pattern harder, as references wouldn't necessarily have `id` any more. This isn't a show stopper, as `equivalent` would also imply the reference.

So I think MUST have an `id` and MAY also have `equivalent` is the right way to go. Worst case scenario, an implementation could skolemize all those blank nodes and be compliant (at the expense of query performance, as above). It encourages implementors to sort out their data and identify the things they actually want to talk about. We can also pursue AAT and other commonly used vocabularies and request more discrete identities for things ... and if we can't get them, we can always mint our own.

To Dave's point about deprecation: We're pre 1.0 still, so have the freedom to change /everything/. To mark something as deprecated in 1.0 would be very weird and we should instead get off the fence and make a decision. Instead, I would say that if it proves to be a real challenge when we have more implementations, then we can deprecate in a 1.1, and remove in 2.0.

**Comment by beaudet** [comment]
Agree on it being silly to deprecate in 1.0, but if we're going to allow metatyping with id, doesn't that make equivalent redundant?  Wouldn't it be just as easy to convert metatypes to blank nodes + equivalents on ingest?

Is there a best practice to suggest to maximize value and minimize the risk of metatyping external ids?

For example, use CAUTION with metatypes:
* you SHOULD not further classify Types with external IDs
* you SHOULD instead first look <here> on the linked art site for common pre-minted "metatypes" such as the enhanced material statement classification to avoid having to use a metatype directly in the first place
* barring that, you SHOULD mint your own classification terms and apply an appropriate equivalent statement in addition to the id of the term.  That will best prepare you for a potential future change in the event that metatyping with ids is deprecated.

**Comment by beaudet** [comment]
As it so happens, I'm generating Concepts for our public data set right now and I have a candidate representation to review.  


The TMS data model for concepts has a term type and a cross reference type associated with terms applied to art entities.  I'm not sure yet exactly how relevant the term types are to a linked data expression.  The values for Term Type are as follows, suggesting to me, at least, that this assignment could potentially indicate that an additional metatype for the term could be applied, but that generally, this is a detail that can probably be skipped for now. 

count      mnemonic     description
1007436	TERM	Descriptor
192162	ALT	Alternate Term
51701	UF	Use For Term
24518	MISC	Miscellaneous Source Equivalent
22804	STE	Search Term Equivalent
17306	LINK	LCSH Link
8778	RT	Related Term
1451	POSTCO	Post Coordinated Terms
1031	UK	British Equivalent
641	UKALT	Alternate British
231	ISO	3-Letter ISO Code
51	POSTAL	Postal Code

There are also cross reference types as follows which, to me, appear to follow the pattern for metatypes much more clearly than the term types listed above.  In fact, some of these metatypes suggest that the term is actually better positioned in other parts of the graph such as place of birth, etc.  I *think* it's probably ok to list all of these under classified_as as well as pulling out specific metatypes for populating other parts of the graph.

count      cross_reference_type     
166092	School
165580	Keyword
73712	Theme
70548	Media
69887	Technique
55822	Support
47367	Scope
37536	Color
32841	Candidate
24795	Gender
19977	IAD Object Name
18238	IAD Microfiche Catalogue Image
18069	Descriptor
17696	IAD Locale
17617	IAD Class
15435	Loan Object Type
12563	Place Executed
11417	Style
9840	Photography Format
9568	Birthplace
9051	Active Place
9012	Systematic Catalogue Volume
8096	Exhibition History Citation
7847	Collection Place
7845	Publication Place
6813	Death Place
6500	Collection
5983	Production Location
3711	Group Name
2371	Housing Needs
2227	Shape
1543	Venue Status
771	Book Format
223	Catalogue Citation
194	(Not Assigned)
178	Object Type
12	Place Depicted (TGN)
1	Place Depicted

As far as implementation, in order to avoid making statements about the Getty AAT terms directly in our data, I've minted a URI for each metatype as its own concept.  In the example below, the NGA's "Active Place" term has the ID listed below.  It is equivalent to the Getty AAT place types term and is further classified with active (professional) to suggest that it's a place where professional activity takes place which seems perfectly reasonable to me.  Since a new ID is minted for each local metatype and the equivalent and classified_as only appear in the Concept entity rather than other entities that make use of this term, there is never a danger of making statements about Getty terms.

```
"id": "https://id.nga.gov/895c8031-4529-4349-863c-14645e308bf5",
    "type": "Type",
    "_label": "Active Place",
    "identified_by": [
        {
            "type": "Name",
            "content": "Active Place",
            "classified_as": [
                {
                    "id": "aat:primaryName_term",
                    "type": "Type",
                    "_label": "primary name"
                }
            ]
        }
    ],
    "classified_as": [
        {
            "id": "https://vocab.getty.edu/aat/300393177",
            "type": "Type",
            "_label": "active (professional function)"
        }
    ],
    "equivalent": [
        {
            "id": "https://vocab.getty.edu/aat/300435109",
            "type": "Type",
            "_label": "place types"
        }
    ]
}
```

Does this look ok to everyone?
